### PR TITLE
Master

### PR DESCRIPTION
**Pull request by v-haiboz**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific langauge SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following apply to the PR so that label “WaitForARMFeedback” will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit (including refactoring) and then push new changes, including version updates, in separate commits.

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by zhenglaizhang** [comment]
Default branch has been switched to main, I have updated the PR to target main

**Comment by jhendrixMSFT** [comment]
I assume these definitions were copied so they could be changed from what's in `common.json`.  What are the changes?

**Comment by daniel-rocha** [comment]
@zhenglaizhang Can you please review @jhendrixMSFT 's question above and explain the problem you are having with autorest and the common, separated GeoJSON definition? Thanks

File: specification/maps/data-plane/Creator/preview/2.0/wfs.json [hunk 1]
```diff
@@ -734,6 +734,156 @@
...
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/GeoJsonObject"
+        }
+      ]
+    },
+    "GeoJsonFeatureCollection": {
+      "description": "A valid `GeoJSON FeatureCollection` object type. Please refer to [RFC 7946](https://tools.ietf.org/html/rfc7946#section-3.3) for details.",
+      "x-ms-discriminator-value": "FeatureCollection",
```
Comment (jhendrixMSFT) [inline specification/maps/data-plane/Creator/preview/2.0/wfs.json:862]: This has the same value as `ExtendedGeoJsonFeatureCollection` which is invalid (all discriminator values must be unique).  Can the value for `ExtendedGeoJsonFeatureCollection` get changed to something like `ExtendedFeatureCollection` (assuming that's unique).

**Review by jhendrixMSFT — CHANGES_REQUESTED** [review]

File: specification/maps/data-plane/Creator/preview/2.0/wfs.json (hunk 1 above)
Comment (v-haiboz) [reply specification/maps/data-plane/Creator/preview/2.0/wfs.json:862]: We have written a tool to fix this issue. There is no need to modify the content, so I closed it.

Comment (v-haiboz) [reply specification/maps/data-plane/Creator/preview/2.0/wfs.json:862]: We have written a tool to fix it.There is no need to modify the content, so I closed this pr.

Comment (jhendrixMSFT) [reply specification/maps/data-plane/Creator/preview/2.0/wfs.json:862]: Does this mean that the duplicate discriminator values will be fixed in some other PR?

Comment (daniel-rocha) [reply specification/maps/data-plane/Creator/preview/2.0/wfs.json:862]: @jhendrixMSFT the duplication was added in this PR to work around an error in doc generation. The main branch shouldn't contain it. 

`
$ find . | grep json | xargs grep x-ms-discriminator-value
./Common/preview/1.0/common.json:      "x-ms-discriminator-value": "Point",
./Common/preview/1.0/common.json:      "x-ms-discriminator-value": "MultiPoint",
./Common/preview/1.0/common.json:      "x-ms-discriminator-value": "LineString",
./Common/preview/1.0/common.json:      "x-ms-discriminator-value": "MultiLineString",
./Common/preview/1.0/common.json:      "x-ms-discriminator-value": "Polygon",
./Common/preview/1.0/common.json:      "x-ms-discriminator-value": "MultiPolygon",
./Common/preview/1.0/common.json:      "x-ms-discriminator-value": "GeometryCollection",
./Common/preview/1.0/common.json:      "x-ms-discriminator-value": "Feature",
./Common/preview/1.0/common.json:      "x-ms-discriminator-value": "FeatureCollection",
./Creator/preview/2.0/featurestate.json:      "x-ms-discriminator-value": "boolean",
./Creator/preview/2.0/featurestate.json:      "x-ms-discriminator-value": "number",
./Creator/preview/2.0/featurestate.json:      "x-ms-discriminator-value": "string",
./Creator/preview/2.0/wfs.json:      "x-ms-discriminator-value": "FeatureCollection",

`

Comment (v-haiboz) [reply specification/maps/data-plane/Creator/preview/2.0/wfs.json:862]: Please ignore this pr. now it does not intend to merge into the main branch. I have closed it.

Comment (jhendrixMSFT) [reply specification/maps/data-plane/Creator/preview/2.0/wfs.json:862]: The duplicates are outside this PR in main now.
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/maps/data-plane/Creator/preview/2.0/wfs.json#L687

https://github.com/Azure/azure-rest-api-specs/blob/main/specification/maps/data-plane/Common/preview/1.0/common.json#L503

Comment (jhendrixMSFT) [reply specification/maps/data-plane/Creator/preview/2.0/wfs.json:862]: ExtendedGeoJsonFeatureCollection is composed of GeoJsonFeatureCollection and they both have the same discriminator value which is malformed.  I haven't checked if there are other instances of this.

Comment (daniel-rocha) [reply specification/maps/data-plane/Creator/preview/2.0/wfs.json:862]: @jhendrixMSFT good catch. I will review and open separate PRs to fix this. Thanks!

Comment (v-haiboz) [reply specification/maps/data-plane/Creator/preview/2.0/wfs.json:862]: cc @zhenglaizhang

Comment (ambientlight) [reply specification/maps/data-plane/Creator/preview/2.0/wfs.json:862]: @daniel-rocha, @jhendrixMSFT: unfortunately there is no different discriminator in ExtendedGeoJsonFeatureCollection, this entity is non-standard, it is a geojson feature collection that contains 3 more non-standard geojson properties (ontology/numberReturned/links), so we won't be able to differentiate it based on the type.

**Pull request closed without merging**